bionmr_utils/md/process/select.py

```diff
@@ -19,67 +31,60 @@ def get_methyl_vectors(frame: Frame) -> List[Tuple[Atom, Atom]]:
                 ('ILE', 'CG2'): ['CG2', 'HG21'],
                 ('MET', 'CE1'): ['CE1', 'HE1']}
 
-    atom_pairs = []
-
-    for r in frame.asResidues:
-        for atom in r.asAtoms:
-            if (r.name.str, atom.name.str) in CH3_dict.keys():
-                C = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][0])]
-                H = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][1])]
-                atom_pairs.append((C, H))
-
-    return atom_pairs
+    C_ids = []
+    H_ids = []
 
+    for r in frame.residues:
+        for atom in r.atoms:
+            if (r.name, atom.name) in CH3_dict.keys():
+                C_ids.append(r[CH3_dict[(r.name, atom.name)][0]].id)
+                H_ids.append(r[CH3_dict[(r.name, atom.name)][1]].id)
 
-def get_NH_vectors(frame: Frame) -> List[Tuple[Atom, Atom]]:
-    """
+    C_atoms = frame.atoms.filter(aId.is_in(set(C_ids)))
+    H_atoms = frame.atoms.filter(aId.is_in(set(H_ids)))
+    atom_pairs_selection = (C_atoms, H_atoms)
 
-    :param frame: Frame
-    :return: list of all backbone atom pairs of given frame.
-    """
-    atom_pairs = []
+    return atom_pairs_selection
 
-    for r in frame.asResidues:
-        try:
-            atom_pairs.append((r[AtomName("N")], r[AtomName("H")]))
-        except pyxmolpp2.polymer.OutOfRangeResidue:
-            pass
 
-    return atom_pairs
+class UnpairedElectronSelection:
+    def __init__(self, nitrogen_selection, oxygen_selection):
+        self.nitrogen_selection = nitrogen_selection
+        self.oxygen_selection = oxygen_selection

```

Comment:
We have just one pair of NO atoms, therefore returned coords.values will be of shape (1,3), not (N,3) as for hydrogen atoms. It still might work due to [numpy broadcasting](https://numpy.org/doc/stable/user/basics.broadcasting.html), but still this is a bit smelly.